### ISSUE TITLE
munit should not need to be installed.

### ISSUE THREAD

**Issue by nanjizal** [vegardit/haxe-doctest#5]
**munit** should not need to be installed but it complains if not listed as a *-lib*, I don't have it installed, I don't want to install it as I am fine using std or if needed fancy would explore tink.

around *line 99* of **DocTestGenerator** needs amends not sure of detail, 
but you can use ```#if munit```

```haxe
    // generate a new testMethod if required
    if (testMethodAssertions.length == MAX_ASSERTIONS_PER_TEST_METHOD ||
     Std.is(doctestAdapter, HaxeUnitDocTestAdapter) 
    #if munit
       || Std.is(doctestAdapter, MUnitDocTestAdapter) 
    #end
```
then at end of the file something like...
``` haxe
     static function getDocTestAdapter():DocTestAdapter {
        var clazz:ClassType = Context.getLocalClass().get();
        while (true) {
            if (clazz.module == "hx.doctest.DocTestRunner") return new TestrunnerDocTestAdapter();
            if (clazz.module == "haxe.unit.TestCase") return new HaxeUnitDocTestAdapter();
            if (clazz.module == "utest.Test") return new HaxeUtestDocTestAdapter(); // need this new adaptor
            #if tink_testrunner // to prevent "Type not found : tink.testrunner.Case" in TinkTestrunnerDocTestAdapter when tink_testrunner is not present
            if (clazz.module == "tink.testrunner.Suite") return new TinkTestrunnerDocTestAdapter();
            #end
            if (clazz.superClass == null) break;
            clazz = clazz.superClass.t.get();
        }

        // if no known super class was found, we expect it to be a MUnit test case
        #if munit
          return new MUnitDocTestAdapter();
        #else
          throw 'failed to find suitable TestAdapter';
          return null;
        #end
    }
```

**Comment by sebthom** [comment]
I cannot reproduce this. I am using haxedoc-test without a library reference to munit for example in [haxe-concurrent](https://github.com/vegardit/haxe-concurrent/blob/master/tests.hxml) or [haxe-strings](https://github.com/vegardit/haxe-strings/blob/master/tests.hxml).

Can you show my the full error including stacktrace please.

**Comment by nanjizal** [comment]
I use this code as per the documentation.
```haxe
@:build(hx.doctest.DocTestGenerator.generateDocTests())
```
that calls
```
var doctestAdapter = getDocTestAdapter();
```
So if I don't have
```
haxe.unit.TestCase
tink.testrunner.Suite
utest.ITest
```
it defaults to try to use munit an errors because it's not testing by string it's testing by name.
you don't get the error in your projects because you have set them up with one of these.
As explained this is because I am using haxe4 and not using and compat3 stuff and need utest adaptor.

**Comment by sebthom** [comment]
I just released haxe-doctest 1.3.0 which additionally supports utest. See https://github.com/vegardit/haxe-doctest#doctest-with-utest

**Comment by nanjizal** [comment]
Oh that's brilliant thanks 👍 🥇
